scripts/factory-agent.sh

```
@@ -1372,6 +1405,196 @@ ensure_beta_build_root() {
+
+FACTORY_LANE_STALE_MIN="${FACTORY_LANE_STALE_MIN:-120}"
+if ! [[ "$FACTORY_LANE_STALE_MIN" =~ ^[1-9][0-9]*$ ]]; then
+  echo "WARNING: invalid FACTORY_LANE_STALE_MIN='$FACTORY_LANE_STALE_MIN'; defaulting to 120" >&2
+  FACTORY_LANE_STALE_MIN=120
+fi
+
+# Whole minutes since an ISO-8601 UTC timestamp, or a huge number if it can't be
+# parsed (an unparseable stamp must not read as "just dispatched" and suppress a
+# lane for ever). BSD date on macOS; -j -f parses rather than sets.
+iso_age_min() {
+  local iso="$1" then now
```

Shellcheck (0.11.0)

[warning] Use semicolon or linefeed before 'then' (or quote to make it literal).

(SC1010)